src/java/fr/paris/lutece/plugins/forms/web/admin/FormQuestionJspBean.java

```diff
@@ -560,6 +582,51 @@ public String getConfirmRemoveComposite( HttpServletRequest request )
 
         return redirect( request, strMessageUrl );
     }
+    
+    private boolean isCompositeHasResponses( int nIdEntry )
+    {
+    	boolean isCompositeHasResponses = false ;
+    	
+    	if ( CompositeDisplayType.QUESTION.getLabel( ).equalsIgnoreCase( _formDisplay.getCompositeType( ) ) && nIdEntry > 0 )
+        {
+    	    ResponseFilter responsefilter = new ResponseFilter( );
+            responsefilter.setIdEntry( nIdEntry );
+            
+            isCompositeHasResponses = isExisteQuestionFilled( responsefilter );
+        }
+    	else if ( CompositeDisplayType.GROUP.getLabel( ).equalsIgnoreCase( _formDisplay.getCompositeType( ) ) )
+        {
+    	    List<Question> questionsList = new ArrayList<>( );
+            ICompositeDisplay composite = _formService.formDisplayToComposite( _formDisplay, null, 0 );
+            composite.addQuestions( questionsList );
+        	
+            if ( CollectionUtils.isNotEmpty( questionsList ) )
+            {
+        	    List<Integer> idEntryList = questionsList.stream( ).map( Question::getIdEntry ).collect( Collectors.toList( ) );
+        	    ResponseFilter responsefilter = new ResponseFilter( );
+        	    responsefilter.setListIdEntry( idEntryList );
+        		
+        	    isCompositeHasResponses = isExisteQuestionFilled( responsefilter );
+            }
+        }
+    	
+    	return isCompositeHasResponses;
+    }
+    
+    private boolean isExisteQuestionFilled( ResponseFilter responsefilter )
+    {
+    	boolean isExisteQuestionFilled = false ;
+    	
+    	List<Response> responseList = _formService.getResponseList( responsefilter );
```

Comment:
Use directly  ResponseHome.getResponseList( responseFilter );
(FormHome and StepHome are already used)

maybe the instance variable _form should be used ? (without forgetting to test that it is still  the right form)

Reply:
The change requested has been implemented